website/docs/reference/datasources/rest-api.md

```diff
@@ -0,0 +1,145 @@
+:::
+
+## Query REST API
+
+The following section is a reference guide that provides a description of the parameters to create REST API queries.
+
+<figure>
+  <img src="/img/restapi-query-config.png" style= {{width:"100%", height:"auto"}} alt="Configuring a REST API query."/>
+  <figcaption align = "center"><i>Configuring a REST API query.</i></figcaption>
+</figure>
+
+<dl>  
+  <dt><b>Method and URL</b></dt>
+  <dd>Sets the REST method and endpoint for the query.</dd>
+</dl>
+
+<dl>  
+  <dt><b>Headers</b></dt>
+  <dd>Sets key/value pairs to send in the request header.</dd>
+  <dd><em>To learn how to set up dynamic headers, visit and fork this <a href="https://app.appsmith.com/applications/6200ac292cd3d95ca414dc4f/pages/624eda0551a8863d6c406760">sample app</a></em>.</dd>
+</dl>
+
+<dl>
+  <dt><b>Params</b></dt>
+  <dd>Sets key/value pairs to send as part of the request.</dd>
+</dl>
+
+<dl>
+  <dt><b>Body</b></dt>
+  <dd><i>Options:</i>
+    <ul>
+      <li><b>None:</b> Sends no body.</li>
+      <li><b>JSON:</b> Expects a JSON object to send as the body.</li>
+      <li><b>FORM_URLENCODED:</b> Expects key/value pairs to be encoded into FORM_URLENCODED format as the body.</li>
+      <li><b>MULTIPART_FORM_DATA:</b> Expects key/value pairs with a data type to be encoded into MULTIPART_FORM_DATA format as the body.</li>
+      <li><b>RAW:</b> Expects raw binary file data to be sent as the body.</li>
+    </ul> 
+  </dd>  
+</dl>
+
+<dl>
+  <dt><b>Pagination</b></dt>
+  <dd><i>Options:</i>
+    <ul>
+      <li><b>None:</b> Doesn't use any pagination.</li>
+      <li><b>Paginate with Table Page No:</b> Use when your API expects an offset or increment value to determine which set of records to return. Follow the instructions that appear on the platform, or see <a href="/reference/widgets/table?current-edition=Offset_edition#server-side-pagination">Offset-based pagination</a> for more information.</li>
+      <li><b>Paginate with Response URL:</b> Use when your API returns cursor values to page through data. The <b>Previous URL</b> and <b>Next URL</b> fields expect the cursor values from the query response. For more information, see <a href="/reference/widgets/table?current-edition=Cursor#server-side-pagination">Cursor-based pagination</a>.</li>
+    </ul>
+  </dd>
+</dl>
+
+<dl>
+  <dt><b>Authentication</b></dt>
+  <dd><em>Use the datasource configuration page to update Authentication settings instead of the query editor.</em></dd>
+</dl>
+
+<dl>
+  <dt><b>Settings</b></dt>
+  <dd>Contains a number of settings to modify the behavior of your query's execution. For reference on each setting, see <a href="/core-concepts/data-access-and-binding/querying-a-database/query-settings">Query Settings</a>.</dd>
+</dl>
+
+---
+
+Below are examples of passing body data in several formats:
+
+<VideoEmbed host="youtube" videoId="znaaDiQbAS8" title="How to pass parameters to an API call" caption="How to pass parameters to an API call"/>
```

Comment:
this is a how to guide. lets link to this video (don't embed) where you are describing the fields